common/transport/coap/src/main/java/org/thingsboard/server/transport/coap/efento/CoapEfentoTransportResource.java

```diff
@@ -272,9 +333,251 @@ private List<EfentoMeasurements> getEfentoMeasurements(MeasurementsProtos.ProtoM
+        values.addProperty("counter_of_confirmable_messages_attempts", runtimeInfo.getMessageCounters(0));
+        values.addProperty("counter_of_non_confirmable_messages_attempts", runtimeInfo.getMessageCounters(1));
+        values.addProperty("counter_of_succeeded_messages", runtimeInfo.getMessageCounters(2));
+        values.addProperty("min_battery_mcu_temp", runtimeInfo.getMinBatteryMcuTemperature());
+        values.addProperty("min_battery_voltage", runtimeInfo.getMinBatteryVoltage());
+        values.addProperty("min_mcu_temp", runtimeInfo.getMinMcuTemperature());
+        values.addProperty("runtime_errors", runtimeInfo.getRuntimeErrorsCount());
+        values.addProperty("up_time", runtimeInfo.getUpTime());
+
+        return new EfentoTelemetry(System.currentTimeMillis(), values);
+    }
+
+    private JsonObject getEfentoConfiguration(ConfigProtos.ProtoConfig config) {
+        JsonObject values = new JsonObject();
+
+        values.addProperty("hash", config.getHash());
+        values.addProperty("serial", CoapEfentoUtils.convertByteArrayToString(config.getSerialNumber().toByteArray()));
+        values.addProperty("timestamp", getDate(config.getHashTimestamp()));
+        values.addProperty("current_time", getDate(config.getCurrentTime()));
+        values.addProperty("apn", getString(config.getApn(), "\u007F", "Automatic"));
+        values.addProperty("token_config", getTokenConfigString(config.getCloudTokenConfig()));
+        values.addProperty("data_server", config.getDataServerIp() + ":" + config.getDataServerPort() +
+                "/" + config.getDataEndpoint());
+        values.addProperty("update_server", config.getUpdateServerIp() + " Coap port " +
+                config.getUpdateServerPortCoap() + " Udp port " + config.getUpdateServerPortUdp());
+        values.addProperty("device_info_endpoint", config.getDeviceInfoEndpoint());
+        values.addProperty("configuration_endpoint", config.getConfigurationEndpoint());
+        values.addProperty("time_endpoint", config.getTimeEndpoint());
+        values.addProperty("transfer_limit", getStringValue(config.getTransferLimit(), 65535, "Disable"));
+        values.addProperty("plmn_selection", getPlmnSelection(config.getPlmnSelection()));
+        values.addProperty("supervision_period", getStringValue(config.getSupervisionPeriod(), 0xFFFFFFFF, "Functionality disabled"));
+        values.addProperty("modem_bands_mask", config.getModemBandsMask());
+        values.addProperty("ACK", getStringValue(config.getAckInterval(), 0xFFFFFFFF, "Always request ACK"));
+        values.addProperty("ble_tx_power_level", config.getBleTxPowerLevel());
+        values.addProperty("channel_types", config.getChannelTypesList().stream().map(Enum::toString).collect(Collectors.joining(",")));
+        values.addProperty("rules", config.getRulesList().stream().map(protoRule -> protoRule.getCondition().toString()).collect(Collectors.joining(",")));
```

Comment:
Please ask @ShvaykaD how to convert the proto object to JSON. I believe we have a lot of boilerplate code that we don't need. Also, I don't like the conversion of RulesList and other objects to String. We should convert them to JSON array or JSON object